### PR TITLE
ensure we emit a rename command for renamed files

### PR DESCRIPTION
**Pull request by mgallien**
Signed-off-by: Matthieu Gallien <matthieu.gallien@nextcloud.com>

### CODE REVIEW

File: src/libsync/propagateremotemove.cpp [hunk 1]
```diff
@@ -254,6 +255,14 @@ void PropagateRemoteMove::finalize()
             newItem._size = oldRecord._fileSize;
         }
     }
+
+    const auto targetFile = propagator()->fullLocalPath(_item->_renameTarget);
```
Comment (allexzander) [inline src/libsync/propagateremotemove.cpp:259]: @mgallien Could you explain what happens here and why?

Comment (mgallien) [reply src/libsync/propagateremotemove.cpp:259]: usually we only rename files on the server that also exists on local sync folder
in my case I would like to do the rename before downloading the file
this is why the local file may not exist even if we rename it on server

**Review by camilasan — APPROVED** [review]

**Review by allexzander — APPROVED** [review]
Many thanks!

**Pull request merged**